frontend/src/pages/projects/projectSharing/ProjectSharingTableRowAdd.tsx

```diff
@@ -54,6 +54,7 @@ const ProjectSharingTableRowAdd: React.FC<ProjectSharingTableRowPropsAdd> = ({
           <Split>
             <SplitItem>
               <Button
+                data-testid={`save-button ${type}`}
```

Comment:
This button test id conflicts with the the one in `ProjectSharingTableRow` because it is accessed without being scoped. If the role is named `User` then it conflicts with the `type` here also being the value `User`. I suggest using different test IDs in this case to distinguish the different behavior of editing an existing entry vs adding a new one.

```suggestion
                data-testid="save-new-button"
```